src/Connection/Http1Connection.php

```diff
@@ -225,14 +258,17 @@ private function doRead(
                             }
                         }
 
-                        if ($this->shouldCloseSocketAfterResponse($response) || $parser->getState() === Http1Parser::BODY_IDENTITY_EOF) {
-                            $this->socket->close();
+                        if ($timeout = $this->determineKeepAliveTimeout($response)) {
```

Comment:
We need to keep the `$parser->getState() === Http1Parser::BODY_IDENTITY_EOF` check here.